lib/Dialect/Torch/Transforms/RefineTypes.cpp

```diff
@@ -750,7 +753,8 @@ static void fillInDTypeGivenDTypeIntAndInputDType(ValueKnowledge &knowledge,
                                                   Value dtype,
                                                   Type inputDType) {
   int64_t dtypeInt;
-  if (dtype.getType().isa<Torch::NoneType>())
+  if (dtype.getType().isa<Torch::NoneType>() ||
+      dtype.getType().isa<OptionalType>())
```

Comment:
actually for `OptionalType` we can't deduce the `dtype` at compile time because the value can be either None or not None at runtime. It's correct to return without filing in `knowledge.dtype` for `OptionalType`

Reply:
Thanks!